colossalai/zero/low_level/bookkeeping/tensor_bucket.py

```diff
@@ -65,12 +65,12 @@ def unflatten_and_copy(self, flat_tensor):
 
     def all_gather(self, group=None, fp8_communication: bool = False):
         flat = self.flatten()
-        buffer = torch.empty(flat.numel() * dist.get_world_size(group), device=flat.device, dtype=flat.dtype)
+        buffers = [torch.empty_like(flat) for _ in range(dist.get_world_size(group))]
         if fp8_communication:
-            all_gather_into_tensor_flat_fp8(buffer, flat, output_shape=buffer.shape, group=group)
+            all_gather_fp8(buffers, flat, group=group, fp8_format="e4m3")
         else:
-            dist.all_gather_into_tensor(buffer, flat, group=group)
```

Comment:
and keep this unchanged

Reply:
Previous PR [low-level-fp8](https://github.com/hpcaitech/ColossalAI/pull/5961/files#r1758289979)  changed it from "dist.all_gather" to "dist.all_gather_into_tensor". I just changed it back

Reply:
all_gather_into_tensor has better performance

Reply:
Fixed